numpy/lib/mixins.pyi

```diff
@@ -0,0 +1,62 @@
+from typing import List
+from abc import ABCMeta, abstractmethod
+
+__all__: List[str]
+
+# NOTE: `NDArrayOperatorsMixin` is not formally an abstract baseclass,
+# even though it's relient on subclasses implementing `__array_ufunc__`
+
+class NDArrayOperatorsMixin(metaclass=ABCMeta):
+    @abstractmethod
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs): ...
```

Comment:
[`NDArrayOperatorsMixin`](https://numpy.org/doc/stable/reference/generated/numpy.lib.mixins.NDArrayOperatorsMixin.html) is not technically an abstract baseclass, but in practice it is relient on subclasses implementing `__array_ufunc__`. Let me know if this should be changed or not.

Reply:
typo: relient -> reliant

Reply:
Good catch, fixed as of https://github.com/numpy/numpy/pull/18853/commits/7d6d74a3ac913ea306d5b237112d3f034b996232.